server/src/openfga-yaml-schema.ts

```diff
@@ -1,4 +1,348 @@
-export const OPENFGA_YAML_SCHEMA = {
+import { AuthorizationModel, Condition, TypeDefinition } from "@openfga/sdk";
+import Ajv, { Schema, ValidateFunction, SchemaValidateFunction } from "ajv";
+
+const identifier = "[a-zA-Z0-9]([a-zA-Z0-9_-]*[a-zA-Z0-9])?";
+
+type Tuple = {
+  user: string,
+  relation: string,
+  object: string,
+  condition?: {
+    name: string,
+    context?: { [param: string]: number | string },
+  }
+}
+
+// Errors for tuples validation
+const invalidType = (user: string, types: string[], instancePath: string) => {
+  return {
+    keyword: "valid_tuple",
```

Comment:
Isn't it weird that the keyword is valid_tuple? Shouldn't it be invalid_tuple

Reply:
Think of it like  the keyword `required`. You put it in the schema like `valid_tuple: true`.

User never sees this, and we can change the visible message to what you like.